src/framework/core/js/Fluid.js

```diff
@@ -808,10 +812,14 @@ var fluid = fluid || fluid_2_0_0;
      * @return The supplied argument, recursively frozen
      */
     fluid.freezeRecursive = function (tofreeze) {
-        fluid.each(tofreeze, function (value) {
-            fluid.freezeRecursive(value);
-        });
-        return fluid.isPlainObject(tofreeze) ? Object.freeze(tofreeze) : tofreeze; // IE11 crashes on freeze of non-object
+        if (fluid.isPlainObject(tofreeze)) {
```

Comment:
If you merge the master branch in, `fluid.isPlainObject()` tests both object and array. Should the second argument added with `true` if you don't wanna freeze arrays and `Object.freeze()` is also not able to.